lib/beamchmark/suite/cpu/cpu_task.ex

```diff
@@ -1,64 +1,75 @@
-defmodule Beamchmark.Suite.CPU.CPUTask do
+defmodule Beamchmark.Suite.CPU.CpuTask do
   @moduledoc """
   This module contains the CPU benchmarking task.
-
+  Measurements are performed using [`:cpu_sup.util/1`](https://www.erlang.org/doc/man/cpu_sup.html)
+  Currently (according to docs), as busy processor states we identify:
+    - user
+    - nice_user (low priority use mode)
+    - kernel
   Run example:
   ```
-  CPUTask.start_link()
+  CpuTask.start_link()
   ```
   """
   use Task
 
   alias Beamchmark.Suite.Measurements.CpuInfo
 
-  @type options_t :: [interval: number(), duration: number()]
+  @interfere_timeout 100
+
+  @doc """
 
-  @spec start_link(options_t()) :: Task.t()
-  def start_link(options) do
+  """
+  @spec start_link(cpu_interval :: pos_integer(), duration :: pos_integer()) :: Task.t()
+  def start_link(cpu_interval, duration) do
     Task.async(fn ->
       run_poll(
-        Keyword.get(options, :interval),
-        Keyword.get(options, :duration)
+        cpu_interval,
+        duration
       )
     end)
   end
 
   @spec run_poll(number(), number()) :: {:ok, CpuInfo.t()} | {:err, String.t()}
```

Comment:
This function will never return `{:err, String.t()}`.

Let's check if our types are proper in other places too by runing `mix dialyzer`.
You can ignore warnings related to `Mix.shell.info`